src/explore-education-statistics-frontend/src/services/permalinkService.ts

```diff
@@ -47,22 +50,163 @@ export interface Permalink {
+) => {
+  /**
+   * config filters only contain `values`.
+   * This function remaps the config from the filters in the subjectMeta
+   */
+  let mappedRows = [];
+  let mappedColumns = [];
+
+  const mappedRowGroups: (LocationFilter | CategoryFilter | any)[][] = [];
+  const mappedColumnGroups: (LocationFilter | CategoryFilter | any)[][] = [];
+
+  const initialValue = columns[0].value;
+  // rows/columns can only be TimePeriods / Indicators
+  // eslint-disable-next-line no-restricted-globals
+  if (isNaN(Number(initialValue))) {
+    // if NaN then timePeriod
+    mappedColumns = columns.map(({ value }) => {
+      return fullTableSubjectMeta.timePeriodRange.find(
+        timePeriod => `${timePeriod.year}_${timePeriod.code}` === value,
+      );
+    });
+    mappedRows = rows.map(({ value }) => {
+      return fullTableSubjectMeta.indicators.find(
+        indicator => indicator.value === value,
+      );
+    });
+  } else {
+    mappedRows = rows.map(({ value }) => {
+      return fullTableSubjectMeta.timePeriodRange.find(
+        timePeriod => `${timePeriod.year}_${timePeriod.code}` === value,
+      );
+    });
+    mappedColumns = columns.map(({ value }) => {
+      return fullTableSubjectMeta.indicators.find(
+        indicator => indicator.value === value,
+      );
+    });
+  }
+
+  // rowGroups/columnGroups can only be filters and locations
+  const locationAndFilterGroups: (LocationFilter | CategoryFilter)[][] = [
+    ...Object.values(
+      transformTableMetaFiltersToCategoryFilters(fullTableSubjectMeta.filters),
+    ),
+    fullTableSubjectMeta.locations,
+  ];
+
+  rowGroups.forEach((rowGroup, index) => {
+    let currentIndex = 0;
+    for (let i = 0; i < locationAndFilterGroups.length; i += 1) {
+      if (
+        locationAndFilterGroups[i].find(
+          element => element.value === rowGroup[0].value,
+        )
+      ) {
+        currentIndex = i;
+        break;
+      }
+    }
+
+    mappedRowGroups[index] = rowGroup.map(({ value }) =>
+      locationAndFilterGroups[currentIndex].find(
+        element => element.value === value,
+      ),
+    );
+    locationAndFilterGroups.splice(currentIndex, 1);
+  });
+
+  columnGroups.forEach((columnGroup, index) => {
+    let currentIndex = 0;
+    for (let i = 0; i < locationAndFilterGroups.length; i += 1) {
+      if (
+        locationAndFilterGroups[i].find(
+          element => element.value === columnGroup[0].value,
+        )
+      ) {
+        currentIndex = i;
+        break;
+      }
+    }
+
+    mappedColumnGroups[index] = columnGroup.map(({ value }) =>
+      locationAndFilterGroups[currentIndex].find(
+        element => element.value === value,
+      ),
+    );
+    locationAndFilterGroups.splice(currentIndex, 1);
+  });
+
+  return {
+    columns: mappedColumns,
+    columnGroups: mappedColumnGroups,
+    rows: mappedRows,
+    rowGroups: mappedRowGroups,
+  };
+};
+
+const mapFullTable = (unmappedFullTable: any) => {
```

Comment:
This function's name is a little dubious as it seems to be just mapping the meta. I would expect it to be called something like `mapSubjectMeta` and return only the `subjectMeta`